### ISSUE TITLE
Rothamsted Core: "All Fields" search functionality to landing page dashboard

### ISSUE THREAD

**Issue by aislinnpearson** [Rothamsted-Ecoinformatics/farm_rothamsted#676]
Following this comment [here ](https://github.com/Rothamsted-Ecoinformatics/farm_rothamsted/issues/652#issuecomment-2177266538) in issue #652, one of our scientists has asked for a 'search all fields' option specifically for the experiment module. Having discussed this with @paul121, who says this is possible (although a medium to large task), I thought it's actually a great idea for all the assets and logs as well so I want to create a larger parent tracking issue that breaks it down into a few sub-tasks. A few related issues I can see:

1. Reviewing and updating the current search fields (renaming and removing search fields based on user feedback, as well as possibly changing some of the columns that are visualised also based on that feedback)
2. Adding the 'search all fields' to the Logs, Assets (and maybe Quantities?) as an initial implementation
3. Thinking in more detail about how to implement this in the experiment module - for example, I could see a situation where you are searching all fields, but the information you want is contained in the Design or the Plan instead of the Experiment. In short, I don't think we should rely on users understanding the data model in order to use the search functionality. I wonder if this might be linked to how we visualise the experiment module... especially in terms of making the different entities more intuitive. See issue #559

**Comment by paul121** [comment]
Drupal core search: https://www.drupal.org/docs/8/core/modules/search/overview

**Comment by aislinnpearson** [comment]
Agreed with @paul121 and @mstenta that we would review what is available in Drupal Core and make a decision from there. We also realised this is at least in part a visualisation issue which might be better resolved by creating a long list of experiments/ proposals by year (similar to what was previously in FieldNet).

If it really does become a usability issue, we could look at more complex options.

**Comment by aislinnpearson** [comment]
Comment 
This issue was originally going to be addressed in milestone 2.23.0, and we made a lot of progress in the search functionality with that milestone, but we agreed a 'search all' need more detailed scoping work. For example, how does this work for logs? Does the process differ for logs associated with experiments? Do we want all assets? All Logs? Should this be on the landing page, or the individual search pages? We agreed to see if this request for advanced search functionality comes up as a user issue. Given the progress we have made, @mstenta I also wondered if it was something you'd want to consider adding to FarmOS core?

@paul121 one idea I had after our call is the idea of an 'advanced' search option on the landing page where you can select to search all research, all assets, all logs (or something along these lines). If so, we might also want to be able to search by FarmOS ID (the number at the end of the url). I know this is something we frequently search by/ look for when discussing individual entities or logs. There is of course the work around, but thought I'd leave a note here so we remember to discuss it's inclusion in future.

**Comment by mstenta** [comment]
> Given the progress we have made, @mstenta I also wondered if it was something you'd want to consider adding to FarmOS core?

I love the idea of taking some of these lessons back to farmOS core! Some kind of unified search would be very useful. I've always sort of expected we would focus on integrations with FOSS search engines like Apache Solr or OpenSearch, but if we find that the performance is good with Rothamsted's data, then maybe that isn't necessary, and a simple generic solution could be built for the majority of farmOS users! I will add a note to our core roadmap so we can consider it.
